libflux/flux-core/src/parser/tests/attributes.rs

```diff
@@ -0,0 +1,395 @@
+                                },
+                            ],
+                        },
+                        id: Identifier {
+                            base: BaseNode {
+                                location: SourceLocation {
+                                    start: "line: 2, column: 1",
+                                    end: "line: 2, column: 4",
+                                    source: "foo",
+                                },
+                            },
+                            name: "foo",
+                        },
+                        init: StringLit(
+                            StringLit {
+                                base: BaseNode {
+                                    location: SourceLocation {
+                                        start: "line: 2, column: 7",
+                                        end: "line: 2, column: 10",
+                                        source: "\"a\"",
+                                    },
+                                },
+                                value: "a",
+                            },
+                        ),
+                    },
+                ),
+            ],
+            eof: [],
+        }
+    "#]]
+        .assert_debug_eq(&parsed);
+}
+
+#[test]
+fn parse_attribute_doc_comment() {
+    let mut p = Parser::new(r#"
+// My documentation comment.
+@deprecated
+identity = (x) => x
+"#);
+    let parsed = p.parse_file("".to_string());
+    expect![[r#"
+        File {
+            base: BaseNode {
+                location: SourceLocation {
+                    start: "line: 3, column: 1",
+                    end: "line: 4, column: 20",
+                    source: "@deprecated\nidentity = (x) => x",
+                },
+            },
+            name: "",
+            metadata: "parser-type=rust",
+            package: None,
+            imports: [],
+            body: [
+                Variable(
+                    VariableAssgn {
+                        base: BaseNode {
+                            location: SourceLocation {
+                                start: "line: 4, column: 1",
+                                end: "line: 4, column: 20",
+                                source: "identity = (x) => x",
+                            },
+                            attributes: [
+                                Attribute {
+                                    base: BaseNode {
+                                        location: SourceLocation {
+                                            start: "line: 3, column: 1",
+                                            end: "line: 3, column: 12",
+                                            source: "@deprecated",
+                                        },
+                                        comments: [
+                                            Comment {
+                                                text: "// My documentation comment.\n",
```

Comment:
This result was concerning to me. The code that it represents seems normal, a comment, an attribute, then a variable assignment. But the comment is attached to the first token which gets attached to the attribute which is a base node itself. This seems correct, but also seems like a breaking change for anything consuming doc comments.

Reply:
If we're fine with this change it seems the code to extract the package doc comment is also in rust so we can probably just change that to search for the appropriate comment.